R/ctwas_wrapper.R

```diff
@@ -329,52 +245,50 @@ get_ctwas_meta_data <- function(ld_meta_data_file, regions_table) {
 }
 
 
-#' Data Loader Function to load twas weights and variant names from output of `load_twas_weights` function.
-#' @param twas_weights_data twas weights data generated from `load_twas_weights`
+#' Data Loader Function to load twas weights and variant names from output of `generate_twas_db` function.
+#' @param twas_weights_data twas weights data generated from `generate_twas_db`
 #' @importFrom R6 R6Class
 #' @export
-# Data Loader Function for importing twas weights from `load_twas_weights` function output.
+# Data Loader Function for importing twas weights from `generate_twas_db` function output.
 twas_weights_loader <- R6Class("twas_weights_loader",
   public = list(
     data = NULL,
     initialize = function(twas_weights_data) {
-      self$data <- twas_weights_data # This is the output from load_twas_weights()
+      self$data <- twas_weights_data # This is the output from generate_twas_db()
     },
     get_weights = function() {
       return(weights = self$data$weights)
     },
     get_susie_obj = function() {
-      return(lapply(self$data$susie_results, function(context_data) {
-        context_data$susie_result[c("lbf_variable", "X_column_scale_factors", "mu")]
-      }))
+      return(lapply(self$data$export_twas_weights_db, function(context_data) context_data$susie_weights_intermediate))
```

Comment:
you hard-coded `susie_weights_intermediate`. Variable `susie_obj` was never used?

Reply:
The object names are now added for object name customizable loader